test/E2ETest/Microsoft.Test.E2E.AspNet.OData/BulkOperation/BulkOperationPatchHandlers.cs

```diff
@@ -9,107 +9,148 @@
 using System.Collections.Generic;
 using System.Linq;
 using Microsoft.AspNet.OData;
+using Microsoft.AspNet.OData.Common;
+using Microsoft.AspNet.OData.Extensions;
 using Microsoft.OData.Edm;
-using Microsoft.Test.E2E.AspNet.OData.BulkInsert;
+using Microsoft.OData.UriParser;
 
 namespace Microsoft.Test.E2E.AspNet.OData.BulkOperation
 {
     internal class APIHandlerFactory : ODataAPIHandlerFactory
     {
-        public APIHandlerFactory(IEdmModel model) : base (model)
+        public APIHandlerFactory(IEdmModel model) : base(model)
         {
         }
 
-        public override IODataAPIHandler GetHandler(NavigationPath navigationPath)
+        public override IODataAPIHandler GetHandler(ODataPath odataPath)
         {
-            if(navigationPath != null)
+            if (odataPath != null)
             {
-                var pathItems = navigationPath;
+                int currentPosition = 0;
 
-                int cnt = 0;
-                                    
-                    switch (pathItems[cnt].Name)
-                    {
-                        case "Employees":
-                        {
-                            Employee employee;
-                            string msg;
-                            if ((new EmployeeAPIHandler().TryGet(pathItems[cnt].KeyProperties, out employee, out msg)) == ODataAPIResponseStatus.Success)
-                            {
-                                return GetNestedHandlerForEmployee(pathItems, cnt, employee);
-                            }
-                        }
-                        return null;
-                    case "Companies":
-                        return new CompanyAPIHandler();
+                if (odataPath.Count == 1)
+                {
+                    GetHandlerInternal(odataPath.FirstSegment.Identifier, currentPosition);
+                }
 
-                        default:
-                            return null;
+                List<ODataPathSegment> pathSegments = odataPath.GetSegments();
 
-                    }
-                
+                ODataPathSegment currentPathSegment = pathSegments[currentPosition];
+
+                if (currentPathSegment is EntitySetSegment || currentPathSegment is NavigationPropertySegment || currentPathSegment is SingletonSegment)
```

Comment:
what happens if the `currentPathSegment` is neither of these segment types?

Reply:
This is customer implemented code. They have control on the types of segments they add here.